simtools/simtel/simtel_events.py

```diff
@@ -1,194 +1,253 @@
-                                thisCombinedHist['data'],
-                                hist['data']
-                            )
-
-                    nFiles += int(countFile)
-
-        self._logger.debug('End of reading {} files'.format(nFiles))
+    def loadHeaderAndSummary(self):
+        '''
+        Read MC header from sim_telarray files and store it into _mcHeader.
+        Also fills summaryEvents with energy and core radius of triggered events.
+        '''
 
-        return combinedHists
+        self._numberOfFiles = len(self.inputFiles)
+        keysToGrab = ['obsheight', 'n_showers', 'n_use', 'core_range', 'diffuse', 'viewcone',
+                      'E_range', 'spectral_index', 'B_total']
+        # Dict has to have its keys defined and filled beforehands
+        self._mcHeader = {k: 0 for k in keysToGrab}
+
+        self.summaryEvents = {
+            'energy': np.array([]),
+            'r_core': np.array([])
+        }
+
+        def _areHeadersConsistent(header0, header1):
+            comparison = dict()
+            for k in keysToGrab:
+                value = (header0[k] == header1[k])
+                comparison[k] = value if isinstance(value, bool) else all(value)
+
+            return all(comparison)
+
+        isFirstFile = True
+        numberOfTriggeredEvents = 0
+        for file in self.inputFiles:
+            with SimTelFile(file) as f:
+
+                for event in f:
+                    en = event['mc_shower']['energy']
+                    rc = math.sqrt(
+                        math.pow(event['mc_event']['xcore'], 2)
+                        + math.pow(event['mc_event']['ycore'], 2)
+                    )
 
-    def _plotCombinedHistograms(self, combinedHists, figName):
+                    self.summaryEvents['energy'] = np.append(self.summaryEvents['energy'], en)
+                    self.summaryEvents['r_core'] = np.append(self.summaryEvents['r_core'], rc)
+                    numberOfTriggeredEvents += 1
+
+                if isFirstFile:
+                    # First file - grabbing parameters
+                    self._mcHeader.update({k: copy(f.mc_run_headers[0][k]) for k in keysToGrab})
+                else:
+                    # Remaining files - Checking whether the parameters are consistent
+                    if not _areHeadersConsistent(self._mcHeader, f.mc_run_headers[0]):
+                        msg = 'MC header pamameters from different files are inconsistent'
+                        self._logger.error(msg)
+                        raise InconsistentInputFile(msg)
+
+                isFirstFile = False
+
+        # Calculating number of events
+        self._mcHeader['n_events'] = (
+            self._mcHeader['n_use'] * self._mcHeader['n_showers'] * self._numberOfFiles
+        )
+        self._mcHeader['n_triggered'] = numberOfTriggeredEvents
+        return
+
+    @u.quantity_input(coreMax=u.m)
+    def countTriggeredEvents(self, energyRange=None, coreMax=None):
```

Comment:
This is just a thought, but would it be better or easier to understand if the energy and code-distances arrays be given as input as well?

I only ask cause while reading the code it took me a second to understand. I don't think it makes a difference.

Reply:
Do you mean adding both to the quantity_input decorator?
That would be ideal but this decorator only accepts scalers, not arrays.

I thought of using energyMin and energyMax to fix that, but I think it's conceptually cleaner to keep energyRange and check the units by ourselves.

Reply:
No, I meant providing as input the energy and core distance arrays. Those which are used in this function with `self.summaryEvents['energy']` and `self.summaryEvents['r_core']`. Now that I think about it again, it probably doesn't make a lot of sense since you want to count for this specific instance of the class. So we can probably safely ignore my comment.